stdlib/universe/fill.go

```diff
@@ -223,13 +220,25 @@ func (t *fillTransformation) Process(id execute.DatasetID, tbl flux.Table) error
 
 	var fillValue interface{}
 	if !t.spec.UsePrevious {
-		if tbl.Cols()[colIdx].Type != flux.ColumnType(t.spec.Value.Type()) {
+		if colIdx > -1 && tbl.Cols()[colIdx].Type != flux.ColumnType(t.spec.Value.Type()) {
 			return errors.Newf(codes.FailedPrecondition, "fill column type mismatch: %s/%s", tbl.Cols()[colIdx].Type.String(), flux.ColumnType(t.spec.Value.Type()).String())
 		}
 		fillValue = values.Unwrap(t.spec.Value)
 	}
 
-	table, err := table.StreamWithContext(t.ctx, key, tbl.Cols(), func(ctx context.Context, w *table.StreamWriter) error {
+	// In case of missing fill column, add it to the existing columns
+	tableCols := tbl.Cols()
+	if colIdx < 0 {
+		newCols := make([]flux.ColMeta, len(tableCols), len(tableCols)+1)
+		copy(newCols, tableCols)
+		c := flux.ColMeta{
+			Label: t.spec.Column,
+			Type:  flux.ColumnType(t.spec.Value.Type()),
+		}
+		tableCols = append(newCols, c)
```

Comment:
Right after this line, I would recommend adding this. See my next comment for why I think this ends up being more clear.

```
colIdx = len(tableCols) - 1
```